improver/blending/weighted_blend.py

```diff
@@ -154,6 +155,18 @@ def _create_model_coordinates(self, cubelist):
         for cube in cubelist:
             cube.attributes[self.model_id_attr] = " ".join(model_titles)
 
+    @staticmethod
+    def _remove_blend_time(cube):
+        """If present on input, remove existing blend time coordinate (as this will
+        be replaced on blending)"""
+        try:
+            coord = cube.coord("blend_time")
+        except CoordinateNotFoundError:
+            return cube
+        else:
+            cube.remove_coord(coord)
+            return cube
```

Comment:
I think this could be simpler. Either (this one is my preference):
```
if "blend_time" in [cooord.name() for coord in cube.coords()]:
   cube.remove_coord("blend_time")
return cube
```
or
```
try:
    cube.remove_coord("blend_time")
except SomeError:
    pass
return cube
```
You don't even need the `return cube` statements if you accept that it is modifying in place anyway.

Reply:
True.  Sadly I do need to return the cube for the list comprehension below.  But I'll simplify as you suggest.